### ISSUE TITLE
file watcher for messages

### ISSUE THREAD

**Issue by samuelstroschein** [inlang/monorepo#1519]
## Problem 

Paraglide has no file watcher for messages yet. 

## Proposal 1 - implement watching capabilities into @inlang/sdk. 

The inlang SDK implements a proxy for `readFile` and a watcher. If changes are detected `loadMessages` is re-executed. Due to the reactivity, messages are automatically updated in apps. 

## Proposal 2 - workaround in apps

Apps write their own proxies and handle watching. 

## Pseudocode

```ts
export const withWatcher({ nodeishFs: NodeishFilesystem, onFileChange: () => void }): NodeishFilesystem => {
   // keeping track of already watched file paths
   const watchedFilePath = []
   // proxy seems to make sense because we only want to intercept readFile 
   return new Proxy(nodeisfhFs, () => {
      // on readFile, intercept the call, check if the file is not watched yet 
      if (readFile && watchedFiles.some(path => path === readFile.path) === false){
        // watch the file
        watchedFilePath.push(readFile.path)
        nodeishFs.watch(readFile.path, onFileChange)
      }
   })
} 
``` 

```ts
// reloading the project on all changes seems to make sufficient. 
const fs = watch({ nodeishFs, onFileChange: async () => {
  setMessages(loadMessages())
}})

```

## Additional information 

I strongly favor implementing file watching in the SDK as the feature is needed for @inlang/paraglide-js @inlang/ide-extension and potentially more inlang apps. 

## Blocked by

- [x] #1534

**Comment by samuelstroschein** [comment]
@inlang/ide-extension are you also favoring an implementation in the SDK?

**Comment by openscript** [comment]
If it's in the SDK and we have something to hook into to repaint the decorations and so on, it will be super easy to implement in ide-extension.

**Comment by NilsJacobsen** [comment]
I think to keep it in the SDK was the original idea. And that is still valid. We should not make a second refactor out of it. So let's spec it.

**Comment by NilsJacobsen** [comment]
I can tackle that on Monday.

**Comment by samuelstroschein** [comment]
I think @inlang/lix fs needs to provide a watch API before you can implement the feature.

**Comment by NilsJacobsen** [comment]
Oh I see. Is this blocking paraglide 1.0.0?

**Comment by samuelstroschein** [comment]
We can implement watching inthe paraglide CLI itself as it only runs in node. But, I prefer an SDK implementation. 

@janfjohannes can you add node watching apis to lix nodeish fs?

**Comment by janfjohannes** [comment]
@samuelstroschein   agree that lix should expose fs watch featrues, i added an issue to our roadmap: https://github.com/orgs/inlang/projects/45/views/1?pane=issue&itemId=43069696

**Comment by janfjohannes** [comment]
@felixhaeberle  @samuelstroschein     unasigning myself, as this ticket is tracking the paraglide side, not the lix part

**Comment by felixhaeberle** [comment]
cc @NilsJacobsen I already got it working supporting only message format, but it would probably cool to have a watcher regardless which plugin is used. i18next, json, imf ...

**Comment by NilsJacobsen** [comment]
Done in https://github.com/inlang/monorepo/pull/1610

**Comment by NilsJacobsen** [comment]
Wait let's wait for the vit eplugin

**Comment by samuelstroschein** [comment]
Implemented (vite plugin is another issue)
